tests/Utils/ClearStateTestListener.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace SimpleSAML\Test\Utils;
+
+use PHPUnit_Framework_Test;
+use PHPUnit_Framework_TestSuite;
+
+/**
+ * A PHPUnit test listener that attempts to clear global state and cached SSP configuration between test run
+ */
+class ClearStateTestListener extends \PHPUnit_Framework_BaseTestListener
```

Comment:
As far as I understand listeners in phpunit, they are executed always, right? That has probably an impact in the time taken to run the tests. I was wondering... wouldn't it be easier to create an abstract class that clears the state before/after each test and then have the tests that need a clear state extend that class? Something like:

```php
class ClearStateTest extends PHPUnit_Framework_TestCase
{
    protected function setUp()
    {
        clearState();
    }

    protected function tearDownAfterClass()
    {
        clearState();
    }

    public static function test1()
    {
         ...
    }

    ...
}

class SomeTestClass extends ClearStateTest
{
    ...
}
```